hrms/hr/doctype/salary_withholding/salary_withholding.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+
+
+class SalaryWithholding(Document):
+	pass
+
+
+@frappe.whitelist()
+def get_employee_details(employee):
+	salary_structure_assignment = frappe.db.get_value(
+		"Salary Structure Assignment",
+		{"employee": employee},
+		"salary_structure",
+	)
+
+	payroll_frequency = frappe.db.get_value(
+		"Salary Structure", salary_structure_assignment, "payroll_frequency"
+	)
+
+	employee = frappe.db.get_all(
+		"Employee",
+		filters={"name": employee},
+		fields=[
+			"date_of_joining",
+			"relieving_date",
+			"employee_name",
+			"resignation_letter_date",
+			"notice_number_of_days",
+		],
+	)
```

Comment:
```suggestion
	employee = frappe.db.get_value(
		"Employee",
		employee,
		fields=[
			"date_of_joining",
			"relieving_date",
			"employee_name",
			"resignation_letter_date",
			"notice_number_of_days",
		],
		as_dict=True,
	)
```
`get_all` is for fetching multiple records. We just need a single record here and also know the name of the record so `get_value` is the correct way

This will eliminate employee[0].date_of_joining too because this will directly return the dictionary not a list of dictionaries